### PR TITLE
Added TDEP parser to an updated kALDo main package for merging.

### PR DESCRIPTION
**Pull request by dafolkner**
Here is the TDEP parser that we forgot to merge. I put it in an updated kALDo package so there are no issues in merging.

### CODE REVIEW

**Comment by dafolkner** [comment]
Pytests no longer fail like they were doing, however, the parser is slow and pytest cannot finish before timeout.

File: kaldo/observables/secondorder.py [hunk 1]
```diff
@@ -9,11 +9,190 @@
 from kaldo.controllers.displacement import calculate_second
 import ase.units as units
 from kaldo.helpers.logger import get_logger, log_size
+from pathlib import Path # NEW !
+from ase.geometry import get_distances # NEW !
+import sys # NEW !
```
Comment (gbarbalinardo-tonal) [inline kaldo/observables/secondorder.py:14]: minor, remove the comments `#NEW!`

File: kaldo/observables/secondorder.py [hunk 2]
```diff
@@ -9,11 +9,190 @@
...
+
+    force_constants = np.zeros((len(uc), len(sc), 3, 3))
+
+    with open(fc_file) as fo:
+        n_atoms = int(next(fo).split()[0])
+        cutoff = float(next(fo).split()[0])
+
+        assert n_atoms == n_uc, f"n_atoms == {n_atoms}, should be {n_uc}"
+
+        #print(f".. Number of atoms:   {n_atoms}")
```
Comment (gbarbalinardo-tonal) [inline kaldo/observables/secondorder.py:59]: Remove not needed comments.

File: kaldo/observables/secondorder.py [hunk 3]
```diff
@@ -9,11 +9,190 @@
...
+                    if np.sum(r_diff) < tol:
+                        force_constants[i1, ii, :, :] += phi
+
+    if not reduce_fc or two_dim:
+        force_constants = remap_force_constants(
+            force_constants, uc, sc, symmetrize=symmetrize
+        )
+
+    if two_dim:
+        return force_constants.swapaxes(2, 1).reshape(2 * (3 * n_sc,))
```
Comment (gbarbalinardo-tonal) [inline kaldo/observables/secondorder.py:83]: Is `reshape(2 * (3 * n_sc,))` correct? Should that be `reshape(2 * (3 * n_sc))`?

File: kaldo/observables/thirdorder.py [hunk 4]
```diff
@@ -251,6 +252,115 @@ def load(cls, folder, supercell=(1, 1, 1), format='sparse', third_energy_thresho
...
+                new_ind += int(n_t*15+1)
+                count += 1
+            while count != 0 and new_ind < len(lines):
+                n_t = int(lines[new_ind].split()[0])
+                num_triplets.append(n_t)
+                new_ind += int(n_t * 15 + 1)
+
+
+            coords = []
+            #frcs = []
```
Comment (gbarbalinardo-tonal) [inline kaldo/observables/thirdorder.py:292]: minor, remove comment

File: kaldo/observables/thirdorder.py [hunk 5]
```diff
@@ -251,6 +252,115 @@ def load(cls, folder, supercell=(1, 1, 1), format='sparse', third_energy_thresho
...
+                        second_cell_list.append(R2)
+                        second_cell_id = (list_of_index[:] == R2).prod(axis=1)
+                        second_cell_id = np.argwhere(second_cell_id).flatten()
+                        third_cell_list.append(R3)
+                        third_cell_id = (list_of_index[:] == R3).prod(axis=1)
+                        third_cell_id = np.argwhere(third_cell_id).flatten()
+                        for alpha in range(3):
+                            for beta in range(3):
+                                for gamma in range(3):
+                                    #coords.append((atom_i, alpha, second_cell_id[0], atom_j, beta, third_cell_id[0], atom_k, gamma))
```
Comment (gbarbalinardo-tonal) [inline kaldo/observables/thirdorder.py:338]: Same

**Review by gbarbalinardo-tonal — COMMENTED** [review]
Great work @dafolkner and nicely done adding the unit tests. Have you tried running the tests locally? How long does it take? Any idea to make it faster?

**Pull request closed without merging**